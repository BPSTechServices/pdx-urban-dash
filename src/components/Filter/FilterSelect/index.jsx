import React from "react";
import PropTypes from "prop-types";
import { Row } from "reactstrap";
import Icon from "../../../components/Icon";

export default class FilterSelect extends React.Component {
  static propTypes = {
    callback: PropTypes.func,
    title: PropTypes.string.isRequired
  };

  constructor(props) {
    super(props);
    this.onClick = this.onClick.bind(this);

    this.callback = data =>
      console.log(
        "FilterSelect Default Callback: " + data.category + ", " + data.title
      );
    this.title = "";
  }

  onClick() {
    this.props.callback({
      title: this.props.title,
      category: this.props.category
    });
  }

  render() {
    return (
      <Row>
        <h5
          style={{
            marginBottom: "1rem",
            marginLeft: "2em",
            whiteSpace: "nowrap"
          }}
          onClick={this.onClick}
        >
          {this.props.title + "   "} <Icon type={"x-circle"} size={"sm"} />
        </h5>
      </Row>
    );
  }
}