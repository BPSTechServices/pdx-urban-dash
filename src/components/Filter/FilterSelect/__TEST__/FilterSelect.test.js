import React from 'react';
import { shallow } from 'enzyme';
import FilterSelect from '../index';

const testFunc = () => { console.log(); };

describe('FilterSelect Render', () => {
  test('snapshot', () => {
    const mockCallBack = jest.fn();
    const wrapper = shallow(<FilterSelect title={'testT'} category={'testC'} callback={mockCallBack}/>);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('FilterSelect callback', () => {
	it('Test click event', () => {
		const mockCallBack = jest.fn();
		const select = shallow(<FilterSelect title={'testT'} category={'testC'} callback={mockCallBack}/>)
		select.find('h5').simulate('click');
		expect(mockCallBack).toHaveBeenCalledWith({title: 'testT', category: 'testC'});
	});
});