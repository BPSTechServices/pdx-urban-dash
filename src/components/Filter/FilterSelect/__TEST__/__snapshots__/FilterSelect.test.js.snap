// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterSelect Render snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FilterSelect
    callback={[MockFunction]}
    category="testC"
    title="testT"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <h5
        onClick={[Function]}
        style={
          Object {
            "marginBottom": "1rem",
            "marginLeft": "2em",
            "whiteSpace": "nowrap",
          }
        }
      >
        testT   
         
        <Icon
          size="sm"
          type="x-circle"
        />
      </h5>,
      "tag": "div",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "testT   ",
          " ",
          <Icon
            size="sm"
            type="x-circle"
          />,
        ],
        "onClick": [Function],
        "style": Object {
          "marginBottom": "1rem",
          "marginLeft": "2em",
          "whiteSpace": "nowrap",
        },
      },
      "ref": null,
      "rendered": Array [
        "testT   ",
        " ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "size": "sm",
            "type": "x-circle",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "h5",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <h5
          onClick={[Function]}
          style={
            Object {
              "marginBottom": "1rem",
              "marginLeft": "2em",
              "whiteSpace": "nowrap",
            }
          }
        >
          testT   
           
          <Icon
            size="sm"
            type="x-circle"
          />
        </h5>,
        "tag": "div",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "testT   ",
            " ",
            <Icon
              size="sm"
              type="x-circle"
            />,
          ],
          "onClick": [Function],
          "style": Object {
            "marginBottom": "1rem",
            "marginLeft": "2em",
            "whiteSpace": "nowrap",
          },
        },
        "ref": null,
        "rendered": Array [
          "testT   ",
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "size": "sm",
              "type": "x-circle",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "h5",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
